### ISSUE TITLE
Does not import multiple-codepoint glyphs from SVG fonts as expected

### ISSUE THREAD

**Issue by DermotMacFlann** [fontforge/fontforge#4860]
I'm using FontForge version 20201107 on Windows 11.  I just barely upgraded from an earlier version of FontForge from 2019, which may also be reflected in some of the screenshots enclosed, but 20201107 displays the exact same behavior.

Recently, I've been participating in an effort at Inkscape to improve its SVG Font Editor.
https://gitlab.com/inkscape/inkscape/-/merge_requests/3628
This has come along well now, but to generate other font formats, you still need to import an SVG font file into FontForge and export from there.

Right now, the most outstanding issue is that SVG font glyphs mapped to strings of multiple Unicode codepoints, do not import as such into FontForge.  And while Inkscape itself does not yet properly preview multiple-codepoint glyphs (except for emojis), FontForge seems to import an SVG font's multiple-codepoint emojis improperly, too.

In this case, it is important that the font I make supports custom glyphs for the Marshallese alphabet letters M̧ m̧ N̄ n̄ O̧ o̧.  Each of these letters can currently only be encoded as a letter with a combining diacritic, requiring two codepoints each, which if I understand correctly, necessitates that each of these dedicated glyphs be encoded as custom ligatures for those codepoints.  I wish this could be done automatically each time an SVG font is imported into FontForge.

The SVG font file itself is still unfinished and considered confidential.  I am willing to share it privately with credible contributors upon request.

![Inkscape FontForge screenshot 1](https://user-images.githubusercontent.com/50077641/143071967-c05d4940-d74b-4fe3-a55f-86bc3a2cf6ec.png)
O̧ (Marshallese letter O with combining cedilla) glyph as shown in Inkscape.

![Inkscape FontForge screenshot 2](https://user-images.githubusercontent.com/50077641/143072046-30759f3b-ced6-470f-af2a-256038519f86.png)
🇲🇭 (Flag of the Marshall Islands) emoji as shown in Inkscape.

![Inkscape FontForge screenshot 3](https://user-images.githubusercontent.com/50077641/143072125-a15a92e1-cc94-44ef-a604-55019a77357e.png)
M̧ (M with combining cedilla) glyph as shown imported into FontForge.  (O̧ and other Marshallese letters as well as the flag are similarly affected.)

![Inkscape FontForge screenshot 4](https://user-images.githubusercontent.com/50077641/143072172-04f69482-b442-4487-84d1-f84b3a44ea5c.png)
🇲🇭 (the flag) emoji as shown imported into FontForge.

**Comment by ChiefMikeK** [comment]
It is not a good idea to encode these alphabetical characters (excluding flag sequences) as multi-point. They should be encoded in PUA as single code point and submitted to the UNICODE Consortium.  
- [13037r-cedillas-and-commas-below.pdf - Unicode.org](https://unicode.org/L2/L2013/13037r-cedillas-and-commas-below.pdf )  
 - [Marshallese site:unicode.org - Google Search](https://www.google.com/search?q=Marshallese+site%3Aunicode.org&newwindow=1&sxsrf=AOaemvJCx1zHysLRu4pJavSBfLlGfvMd5Q%3A1637745744785&ei=UASeYbCrL8ri-Aayxo6gAw&ved=0ahUKEwiwk-aq1rD0AhVKMd4KHTKjAzQQ4dUDCA4&uact=5&oq=Marshallese+site%3Aunicode.org&gs_lcp=Cgdnd3Mtd2l6EANKBAhBGAFQzipYrTxg6TxoAnAAeACAAYcBiAGHAZIBAzAuMZgBAKABAqABAcABAQ&sclient=gws-wiz )

**Comment by DermotMacFlann** [comment]
Yes, I'm well aware of these issues.  I helped edit articles on Wikipedia describing it.  It's a well-established issue that has even specifically been addressed in the maintenance of certain fonts like Noto.
https://github.com/googlefonts/noto-fonts/issues/505
This is the standard alphabet for the Marshallese language, and the moment, the **only** proper way to encode these specific letters is multi-point.  There's no two ways about it.  And this isn't even that super-new of an alphabet, either - it was codified in the 1970s.

Io̧kwe. :)

**Comment by ctrlcctrlv** [comment]
I don't know where you're getting your information from @ChiefMikeK, but the Unicode Technical Committee encourages the use of multi-codepoint encoding for Latin orthographies just as it does for all other orthographies. The atomic characters such as ‹É› are historical artifacts which only exist for transparent compatibility with documents encoded in various ISO encodings that were devised before combining character support was widespread.

**Comment by iterumllc** [comment]
If I understand it right your request is to treat the glyphs you import as ligatures which are then mapped via OpenType substitutions to base-accent combinations. 

The normal way this would happen in FontForge is to have the actual glyphs "umapped" (i.e. mapped to -1) and then build up the substution table. It's easy enough to import a glyph unmapped as things are. So the missing functionality boils down to constructing the substitution table, maybe in `liga`, which is usually "on" by default. 

I doubt support for this automatic construction will be added to FontForge "natively" anytime soon -- there are simply too many requests for this or that feature. However, I think it would be easy enough to script via Python. 

My first suggestion is to verify that unmapped glyphs plus a `liga` table does what you want. Try following [these instructions](https://fontforge.org/docs/tutorial/editexample4.html) using the UI with one or two of your ligatures and see if it works after you generate an OpenType font and use it in a target application. If it does we can talk about writing a little script to build up the table from whatever SVG-level data you have. 

(Note that in order to make the substitution table work in FontForge you may (or may not) need the combining-character glyph to exist, even if it will never appear. @ctrlcctrlv will likely know more about this than I do. If so, you can either just "give it width" and leave it blank or copy a version of the accent into it for clarity.)

**Comment by ctrlcctrlv** [comment]
Yes, I've done it before, it does need the character to exist even if it will never appear because otherwise there's nothing to refer to in the substitution table. (I do this in FRB American Cursive.) Also, the typical tag name for this is `ccmp`, not `liga`, but both should work, just `liga` can be turned off and these aren't considered ligatures so I really recommend you use `ccmp`.
